### PR TITLE
update build/run dependencies.

### PR DESCRIPTION
**Pull request by molpopgen**
updates the build/run dependencies missed in #22834

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by molpopgen** [comment]
@bioconda/core -- I'm trying to sort out the build failure here (and in the autobump #22834).  Is there a way to actually see the full build log?  The current output is hiding the error messages:

```
c4081a67e560: Pull complete 
Digest: sha256:7bf2ff10cc23e3a8d9028cd4eec9457de3527b19fbdc2fc64e122b68548b1b36
Status: Downloaded newer image for quay.io/dpryan79/mulled_container:latest
23:14:57 BIOCONDA INFO Considering total of 7334 recipes.
23:14:58 BIOCONDA INFO Constraining to 1 git modified recipes (fwdpy11).
23:14:58 BIOCONDA INFO Processing 1 recipes (fwdpy11).
23:14:58 BIOCONDA INFO Generating DAG
23:14:58 BIOCONDA INFO 1 recipes to build and test: 
fwdpy11
23:14:58 BIOCONDA INFO Determining expected packages for recipes/fwdpy11
Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
23:14:58 CONDA_BUILD WARNING Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
23:14:58 CONDA_BUILD WARNING No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
23:14:58 CONDA_BUILD INFO Adding in variants from internal_defaults
Adding in variants from /home/circleci/project/miniconda/conda_build_config.yaml
23:14:58 CONDA_BUILD INFO Adding in variants from /home/circleci/project/miniconda/conda_build_config.yaml
Adding in variants from /home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/bioconda_utils-conda_build_config.yaml
23:14:58 CONDA_BUILD INFO Adding in variants from /home/circleci/project/miniconda/lib/python3.7/site-packages/bioconda_utils/bioconda_utils-conda_build_config.yaml
Attempting to finalize metadata for fwdpy11
23:15:11 CONDA_BUILD INFO Attempting to finalize metadata for fwdpy11
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
.....23:23:48 BIOCONDA ERROR COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmpz619z5r3/build_script.bash:/opt/build_script.bash -v /home/circleci/project/miniconda/conda-bld/:/opt/host-conda-bld -v /home/circleci/project/recipes/fwdpy11:/opt/recipe -e LANG=en_US.UTF-8 -e HOST_USER_ID=1001 bioconda/bioconda-utils-build-env:latest /bin/bash /opt/build_script.bash
ERROR BIOCONDA:run(626): COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmpz619z5r3/build_script.bash:/opt/build_script.bash -v /home/circleci/project/miniconda/conda-bld/:/opt/host-conda-bld -v /home/circleci/project/recipes/fwdpy11:/opt/recipe -e LANG=en_US.UTF-8 -e HOST_USER_ID=1001 bioconda/bioconda-utils-build-env:latest /bin/bash /opt/build_script.bash

23:23:48 BIOCONDA ERROR BUILD FAILED recipes/fwdpy11
ERROR BIOCONDA:build(140): BUILD FAILED recipes/fwdpy11
23:23:49 BIOCONDA ERROR BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
ERROR BIOCONDA:build_recipes(370): BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
23:23:49 BIOCONDA ERROR BUILD SUMMARY: FAILED recipe recipes/fwdpy11
ERROR BIOCONDA:build_recipes(376): BUILD SUMMARY: FAILED recipe recipes/fwdpy11

Exited with code exit status 1
```

**Comment by dpryan79** [comment]
@mbargull I've seen the above behavior where most of the normal output is surpressed in other recent PRs. Was this an inadvertent side-effect to updating the conda version in the most recent bioconda-utils?

@molpopgen Locally I have a slightly older version of bioconda-utils installed and that reliably displays the normal build information. That won't fix things for this PR, but it's at least a hacky solution.

**Comment by molpopgen** [comment]
@dpryan79 thanks.  The issue now is that Linux is passing but not macos.  I don't have easy access to macos to try anything locally.

File: recipes/fwdpy11/meta.yaml [hunk 1]
```diff
@@ -24,13 +24,17 @@ requirements:
     - tskit >=0.1.5
     - gsl
     - openblas
+    - attrs
```
Comment (mbargull) [inline recipes/fwdpy11/meta.yaml:27]: @molpopgen, I just assumed the `attr` requirement was supposed to be `attrs`. Is that correct?

File: recipes/fwdpy11/meta.yaml [hunk 2]
```diff
@@ -24,13 +24,17 @@ requirements:
...
+    - setuptools_scm
   run:
     - python
     - numpy >=1.10
     - tskit >=0.1.5
     - gsl
     - openblas
     - sparse
+    - attrs
+    - black
```
Comment (mbargull) [inline recipes/fwdpy11/meta.yaml:37]: Just curious: I `black` really required as a `run` dependency? Seems rather unusual, is all.

File: recipes/fwdpy11/meta.yaml (hunk 1 above)
Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: The conda package is called attr or attrs.  The former is from forge, I believe.

File: recipes/fwdpy11/meta.yaml (hunk 2 above)
Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:37]: Yes, it is used to provide a method to pretty-print string representations of classes.

File: recipes/fwdpy11/meta.yaml (hunk 1 above)
Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:27]: Sorry for not giving more context. So, conda-forge has both `attr` and `attrs`. `attrs` is the Python project (https://www.attrs.org/en/stable/) and `attr` is the GNU project http://savannah.nongnu.org/projects/attr.
This being a Python package, I assumed you'd want `attrs` and not the Linux-specific `attr`.

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: https://anaconda.org/anaconda/attrs

https://anaconda.org/conda-forge/attr

However, it looks like the attr version doesn't support macos.  Ugh. On PyPi, both attr and attrs are valid.  That probably explains the macos build failure.

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: > Sorry for not giving more context. So, conda-forge has both `attr` and `attrs`. `attrs` is the Python project (https://www.attrs.org/en/stable/) and `attr` is the GNU project http://savannah.nongnu.org/projects/attr.
> This being a Python package, I assumed you'd want `attrs` and not the Linux-specific `attr`.

Yeah, I just figured that out.  Quite confusing, as you can pip install either name.

Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:27]: Yep, the naming is rather confusing, I agree. On top of that, `pip attr` would probably give you yet *another* package (https://pypi.org/project/attr/) o_O.

Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:27]: (Plus the `attr.s` thing of course...)

File: recipes/fwdpy11/meta.yaml (hunk 2 above)
Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:37]: Ah, okay -- thanks for clarifying!

File: recipes/fwdpy11/meta.yaml (hunk 1 above)
Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: Yes, it does seem like a bit of a joke at times.

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: Well, changing attr to attrs now makes the Linux check fail.  But I can't see the logs to know why.

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: So updating the tskit min version. Makes the checks pass now.

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: Seems like this is okay to merge?

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: Should I change the merge target to master instead of the bump branch?  If so, we can close the autobump.

**Comment by mbargull** [comment]
@BiocondaBot, please fetch artifacts

**Comment by mbargull** [comment]
@BiocondaBot, please merge

File: recipes/fwdpy11/meta.yaml (hunk 1 above)
Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:27]: Yep, looks all good AFAICT. Thanks for the updates!

Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:27]: Also, the "can't see the logs" thing should be fixed now. I had to add a workaround for newer `conda` versions to `bioconda-utils`. Thanks for making us aware of the issue!

Comment (molpopgen) [reply recipes/fwdpy11/meta.yaml:27]: Great!  Thanks for fixing that.  It makes things a lot easier for systems where you cannot replicate things locally.

Comment (mbargull) [reply recipes/fwdpy11/meta.yaml:27]: > cannot replicate things locally.

Most definitely! I hate those things :).

**Pull request merged**